tests/test_basic_ops_dict.py

```diff
@@ -398,17 +443,162 @@ def test_dict_pop_error(cfg: Dict[Any, Any], key: Any, expectation: Any) -> None
             "incompatible_key_type",
             False,
         ),
+        (
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            "FOO",
+            True,
+        ),
+        (
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            None,
+            False,
+        ),
+        # int key type
         ({1: "a", 2: {}}, 1, True),
         ({1: "a", 2: {}}, 2, True),
         ({1: "a", 2: {}}, 3, False),
         ({1: "a", 2: "???"}, 2, False),
+        ({1: "a", 2: "???"}, None, False),
+        ({1: "a", 2: "???"}, "1", False),
+        # (DictConfig({1: "a", 2: "???"}, key_type=int), "1", True),
```

Comment:
Why is this commented out?

Reply:
Oops.

Reply:
Fixed in https://github.com/omry/omegaconf/pull/484/commits/85801f4e48f2401e4bfa4d090b4f37fbef38d887